Create documentation for component and related packages (#7)
kyma-incubator/documentation-component: docs/props/sources.md

# `sources` property

## Overview

The `sources` property is required and contains sources for component. 

## Types

The `sources` property is array type of `SourceWithOptions` and/or `SourceGroupWithOptions`. See characteristic of those structures below.

``` ts
interface SourceWithOptions {
  source: {
    type: string;               // type of document, e.g.: "md"

>     rawContent: string | any;   // content of document

Reviewer: this type may change after my review of magic's PR, to be adressed later

    data?: any;                 // additional data - extracted or using by plugin or render engine
  };
  pluginsOptions?: {
    name: string;
    options: any; 
  }[];
  renderEngineOptions?: {
    name: string;
    options: any; 
  }[];
};

interface SourceGroupWithOptions {
  sources: SourceWithOptions[];
  pluginsOptions?: {
    name: string;
    options: any; 
  }[];
  renderEngineOptions?: {
    name: string;
    options: any; 
  }[];
};
```

## Passing options

> Single `pluginsOptions` and `renderEngineOptions` are objects with two fields: `name`(`string` type) and `options`(`object` type). For passing `pluginsOptions` and/or `renderEngineOptions` you have to define for what plugin/render engine you want to provider options. Defines using the name of the plugin/render plugin as `name` field, and then provide options. 

Reviewer: ...what plugin/render engine you want to **provide** options. ...

### Concatenating options

> Core of `documentation-component` automatically concatenate options passing globally, for group of sources, and for single source with priorities: 

Reviewer 2: ```suggestion
The core of the Documentation Component automatically concatenates options passed globally, for a group of sources, and for a single source, in this order of priorities: 
```

> - single source,

Reviewer 2: ```suggestion
1. Single source
```

> - group of sources,

Reviewer 2: ```suggestion
2. Group of sources
```

> - options passing globally.

Reviewer 2: ```suggestion
3. Options passed globally
```

> > **NOTE**: For information how to pass options globally for plugins read [section](). For render engine, read [this]() section.

Reviewer 2: ```suggestion
> **NOTE**: For information on how to pass options globally:
>  - for plugins, read [this]() section.
>  - for render engines, read [this]() section.
```

Reviewer 2: Missing links ;)

## Example

> 

Reviewer 2: See an example that demonstrates how to pass options for various source types:
